deno/payloads/v9/channel.ts

```diff
@@ -14,61 +14,91 @@ import type { APIUser } from './user.ts';
 /**
  * Not documented, but partial only includes id, name, and type
  */
-export interface APIPartialChannel {
+export interface APIPartialChannel<T extends ChannelType> {
 	/**
 	 * The id of the channel
 	 */
 	id: Snowflake;
 	/**
-	 * The type of the channel
-	 *
+	 * The name of the channel (2-100 characters)
+	 */
+	name?: string;
+	/**
 	 * See https://discord.com/developers/docs/resources/channel#channel-object-channel-types
 	 */
-	type: ChannelType;
+	type: T;
+}
+
+export type TextBasedChannels =
+	| ChannelType.DM
+	| ChannelType.GroupDM
+	| ChannelType.GuildNews
+	| ChannelType.GuildPublicThread
+	| ChannelType.GuildPrivateThread
+	| ChannelType.GuildNewsThread
+	| ChannelType.GuildStore
```

Comment:
Not text based at all

Reply:
I'm not sure I follow, don't threads news, and stores all use text input in their channels?

Reply:
Pretty sure store channels don’t have any input, it’s just items you can buy

Reply:
The store channel isn't a text based one! :D